Common/util/misc.cpp

```diff
@@ -115,7 +115,7 @@ char *ci_find_file(const char *dir_name, const char *file_name)
     fix_filename_slashes(filename);
   }
 
-  if(directory && filename) {
+  if(directory && filename && !strstr(filename, "..")) {
```

Comment:
I don't think this is a correct check. It's possible to have a file name containing two dots.

Reply:
yeah, but in that case we fallback to the chdir/opendir approach below

Reply:
Could you add a small comment above noting this?

Reply:
sure!

Reply:
thank you )